src/reactivity/unwrap.ts

```diff
@@ -0,0 +1,45 @@
+import { isRef } from './ref';
+import { proxy, isFunction, isObject, isArray } from '../utils';
+import { isReactive } from './reactive';
+
+export function unwrapRefProxy(value: any) {
+  if (isFunction(value)) {
+    return value;
+  }
+
+  if (isRef(value)) {
+    return value;
+  }
+
+  if (isArray(value)) {
+    return value;
+  }
+
+  if (isReactive(value)) {
+    return value;
+  }
+
+  if (!isObject(value)) {
+    return value;
+  }
+
```

Comment:
Should we add `!Object.isExtensible(value)`?

Reply:
Can you provide a case? 

I don't think so, because we don't alter the object, just create a clone of it and unwrap `refs`

Reply:
I think a frozen object should remain untouched (and it will disable Vue's reactivity interpretation inside it, which can improve performance in some applications).

We also run the unwrap before `markRaw`, if we do clone it, #366 will never be covered in this usage.

Reply:
added the condition